server/api/resolvers/requestResolvers.ts

```diff
@@ -0,0 +1,123 @@
+import { Client, ClientInterface } from '../../database/models/clientModel'
+import { Request, RequestInterface } from '../../database/models/requestModel'
+import { RequestType, RequestTypeInterface } from '../../database/models/requestTypeModel'
+
+import { sessionHandler } from '../utils/session'
+
+const requestEmbeddingFromRequest = ( request: RequestInterface ) => {
+    return {
+        _id: request._id,
+        createdAt: request.createdAt ? request.createdAt : null,
+        deletedAt: request.deletedAt ? request.deletedAt : null,
+        fulfilledAt: request.fulfilledAt ? request.fulfilledAt : null
+    }
+}
+
+const swapRequestTypeForRequest = async ( request, oldRequestTypeID, newRequestTypeID, session ) => {
+    const oldRequestType = await RequestType.findById(oldRequestTypeID).session(session)
+    oldRequestType.requests = oldRequestType.requests.filter((requestEmbedding) => {
+        return requestEmbedding._id !== request._id
+    })
+
+    const newRequestType = await RequestType.findById(newRequestTypeID).session(session)
+    newRequestType.requests.push(requestEmbeddingFromRequest(request))
+
+    await oldRequestType.save({ session: session })
+    await newRequestType.save({ session: session })
+}
+
+const requestQueryResolvers = {
+    request: async (_, { _id }, ___): Promise<RequestInterface> => {
+        return Request.findById(_id).exec()
+    },
+    requests: async (_, __, ___): Promise<Array<RequestInterface>> => {
+        return Request.find().exec()
+    },
+    requestsFilter: async (_, { filter, options }, ___): Promise<Array<RequestInterface>> => {
+        return Request.find().exec()
+    },
+}
+
+const requestMutationResolvers = {
+    createRequest: async (_, { request }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => { 
+            return sessionHandler(async (session) => {
+                const newRequestObject = new Request({...request})
+                const newRequest = await newRequestObject.save({ session: session })
+                
+                const requestType = await RequestType.findById(newRequest.requestType).session(session)
```

Comment:
should this be RequestType.findById(newRequest.requestType._id) ?

Reply:
Mongoose model shows Request's requestType is an ID

Reply:
So newRequest.requestType is the ID